File: limiteMEI-api/src/main/java/com/limiteMEI/limiteMEI/dto/pessoa/PessoaDTO.java
package com.limiteMEI.limiteMEI.dto.pessoa;

import com.limiteMEI.limiteMEI.enums.PapelPessoaEnum;
import com.limiteMEI.limiteMEI.enums.BancoEnum;
import com.limiteMEI.limiteMEI.enums.TipoContaFinanceiraEnum;
import com.limiteMEI.limiteMEI.enums.TipoPessoaEnum;
import com.limiteMEI.limiteMEI.enums.UfEnum;
import lombok.*;
import java.util.Set;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PessoaDTO {
    private Long id;
    private TipoPessoaEnum tipoPessoa;
    private String nomeRazaoSocial;
    private String nomeFantasia;
    private String cpfCnpj;
    private String email;
    private String telefone;
    private String telefoneAlternativo;
    private String responsavel;
    private String cep;
    private String endereco;
    private String numero;
    private String complemento;
    private String bairro;
    private String cidade;
    private UfEnum uf;
    private String observacoesComerciais;
    private BancoEnum banco;
    private String agencia;
    private String conta;
    private TipoContaFinanceiraEnum tipoConta;
    private String chavePix;
    private Boolean ativo;
    private Set<PapelPessoaEnum> papeis;
}

File: limiteMEI-angular/src/app/dtos/pessoa/pessoa.dto.ts
import {GridColumn} from '../../shared/utils/directives/grid.column.decorator';import {TipoPessoaEnum,TIPO_PESSOA_LABELS} from '../../enums/tipo.pessoa.enum';import {PapelPessoaEnum} from '../../enums/papel.pessoa.enum';import {BancoEnum} from '../../enums/banco.enum';import {TipoContaFinanceiraEnum} from '../../enums/tipo.conta.financeira.enum';import {UfEnum, UF_LABELS} from '../../enums/uf.enum';
export interface RelatorioPessoaFiltroDTO{nome?:string;documento?:string;tipoPessoa?:TipoPessoaEnum;}
export class PessoaDTO{
 @GridColumn({label:'Código',type:'number',ordem:1,width:'90px'}) id!:number;
 @GridColumn({label:'Nome / Razão social',type:'texto',ordem:2}) nomeRazaoSocial!:string;
 @GridColumn({label:'Tipo',type:'enum',ordem:3,width:'150px',enumLabels:TIPO_PESSOA_LABELS}) tipoPessoa!:TipoPessoaEnum;
 @GridColumn({label:'CPF / CNPJ',type:'documento',ordem:4,width:'190px'}) cpfCnpj?:string;
 @GridColumn({label:'E-mail',type:'texto',ordem:5}) email?:string;
 @GridColumn({label:'Telefone',type:'texto',ordem:6,width:'150px'}) telefone?:string;
 @GridColumn({label:'Cidade',type:'texto',ordem:7,width:'170px'}) cidade?:string;
 @GridColumn({label:'UF',type:'enum',ordem:8,width:'90px',enumLabels:UF_LABELS}) uf?:UfEnum;
 @GridColumn({label:'Ativa',type:'boolean',ordem:9,width:'90px'}) ativo!:boolean;
 nomeFantasia?:string;
 telefoneAlternativo?:string;
 responsavel?:string;
 cep?:string;
 endereco?:string;
 numero?:string;
 complemento?:string;
 bairro?:string;
 observacoesComerciais?:string;
 banco?:BancoEnum;
 agencia?:string;
 conta?:string;
 tipoConta?:TipoContaFinanceiraEnum;
 chavePix?:string;
 papeis:PapelPessoaEnum[]=[];
}
